### PR TITLE
feat(FirstTokenReceivedPopup): Created new popup `FirstTokenReceived`

### LINKED ISSUE #12366: [Wallet - Token management] First community asset details popup
### Description
[Design](https://www.figma.com/file/FkFClTCYKf83RJWoifWgoX/Wallet-v2?type=design&node-id=17159%3A44846&mode=design&t=wHGfj066NK7EizhU-1)

Screenshot 2023-10-06 at 12 54 01

Create the popup using `storybook` and integrate it into the app.

### PR DESCRIPTION
**Pull request by noeliaSD**
Closes #12366

### What does the PR do

- Created new popup component.
- Created related `storybook` page.
- Integration of the popup into the app.

**NOTE:** Out of scope --> Hide token button -->  #13293

### Affected areas

First token received toast

### Screenshot of functionality 

- Storybook:

https://github.com/status-im/status-desktop/assets/97019400/b617de33-7d25-4fb2-bf4d-e3e1b9695552

- App integration (hardcoded code to force the toast is being displayed):

https://github.com/status-im/status-desktop/assets/97019400/e50acd15-1367-4014-8531-81c1a729b9b4

### CODE REVIEW

File: ui/imports/shared/stores/CommunityTokensStore.qml [hunk 1]
```diff
@@ -136,16 +137,18 @@ QtObject {
...
             // ** `isFirst` property will be true if it's the first time the user receives a community asset and  a community collectible
             // ** `tokenType` property will determine if the received minted token is an ERC20 or an ERC720
             // ** `walletAccountName` property will provide the wallet account name where the token was received
+            // ** `symbol` property will provide the token symbol
 
             var isFirst = false
             var tokenType = Constants.TokenType.ERC20
             var walletAccountName = "Status account"
-            root.communityTokenReceived(name, image, communityId, communityName, balance, chainId, txHash, isFirst, tokenType, walletAccountName)
+            var symbol = "NON"
```
Comment (caybro) [inline ui/imports/shared/stores/CommunityTokensStore.qml:150]: This looks wrong, hardcoded symbol?

File: ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    // Community related props:
+    property string communityId
+    required property string communityName
+    required property string communityLogo
+
+    // Token related props:
+    required property string tokenName
+    required property string tokenSymbol
+    required property string tokenImage
+    required property string tokenAmount
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml:29]: Should be an `int` imho

File: ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                onClicked: {
+                    close()
+                    root.hideClicked()
+                }
+            }
+
+            StatusButton {
+                id: acceptBtn
+
+                text: qsTr("Go it!")
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml:140]:
```suggestion
                text: qsTr("Got it!")
```

typo?

File: ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        rightButtons: ObjectModel {
+            StatusFlatButton {
+                id: hideBtn
+
+                visible: false // TODO: #13293
+
+                text: root.isAssetType ? qsTr("Hide this asset") : qsTr("Hide this collectible")
+
+                onClicked: {
+                    close()
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml:132]:
```suggestion
                    root.close()
```

File: ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+                    root.hideClicked()
+                }
+            }
+
+            StatusButton {
+                id: acceptBtn
+
+                text: qsTr("Go it!")
+
+                onClicked: close()
```
Comment (caybro) [inline ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml:142]:
```suggestion
                onClicked: root.close()
```

File: ui/app/mainui/ToastsManager.qml [hunk 6]
```diff
@@ -216,7 +221,24 @@ QtObject {
...
+                var communityId = data.communityId
+                var communityName = data.communityName
+                var tokenType = data.tokenType
+                var tokenName = data.tokenName
+                var tokenSymbol = data.tokenSymbol
+                var tokenImage = data.tokenImage
+                var tokenAmount = data.tokenAmount
+                Global.openFirstTokenReceivedPopup(communityId,
+                                                   communityName,
+                                                   rootChatStore.getCommunityDetailsAsJson(communityId).image,
```
Comment (alexjba) [inline ui/app/mainui/ToastsManager.qml:235]: Can we receive the community image from the backend as well?

**Review by alexjba — COMMENTED** [review]
Looks good in general.
I've added a few questions for the things that are not clear to me.

**Comment by alexjba** [comment]
There's also this question here that might have an impact on this PR. If we can receive toast notifications with the community info missing, we will probably need to handle this case in the UI.

https://github.com/status-im/status-go/pull/4515#issuecomment-1910045200

File: ui/app/mainui/ToastsManager.qml (hunk 6 above)
Comment (alexjba) [reply ui/app/mainui/ToastsManager.qml:235]: @Cuteivist do you plan to expose it from nim, or should we create a separate task for this?

File: ui/imports/shared/stores/CommunityTokensStore.qml (hunk 1 above)
Comment (noeliaSD) [reply ui/imports/shared/stores/CommunityTokensStore.qml:150]: There's a missing backend. There's a comment on top that point to the task: #13250

File: ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml (hunk 2 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Communities/popups/FirstTokenReceivedPopup.qml:29]: Assets can be decimals amounts

File: ui/app/mainui/ToastsManager.qml (hunk 6 above)
Comment (noeliaSD) [reply ui/app/mainui/ToastsManager.qml:235]: @Cuteivist if this pr is merged before you finish with the backend, feel free to update the UI accordingly!

**Comment by noeliaSD** [comment]
> There's also this question here that might have an impact on this PR. If we can receive toast notifications with the community info missing, we will probably need to handle this case in the UI.
> 
> [status-im/status-go#4515 (comment)](https://github.com/status-im/status-go/pull/4515#issuecomment-1910045200)

There's this task to analyze how to behave in this case: https://github.com/status-im/status-desktop/issues/13296

**Comment by noeliaSD** [comment]
Updated PR according to @alexjba and @caybro comments!! Could you please take a second look?

**Review by caybro — APPROVED** [review]
LGTM

**Review by alexjba — APPROVED** [review]
LGTM! Thanks! 👍

**Pull request merged**